### ISSUE TITLE
Authenticated controller support - Ways to support them

### ISSUE THREAD

**Issue by dogtopus** [FeralAI/GP2040#64]
Pretty much just a table listing how to support controllers with authentications.

| Type | Passthrough | Security chip hijacking | Native | Notes |
| --- | --- | --- | --- | --- |
| PS4 | ⭕ | △ | △ | Chip hijacking only works with 3rd party controllers and native requires dumping keys from one's own eligible PS4 controller motherboard. |
| PS5 | ❌ | △ | ❌ | May be possible by using a certain overpriced (and reportedly broke after firmware update) UP5 module. Still f\*\*k $\*\*y 🖕 |
| Xbox 360 | ⭕ | △ | ❌ | Secrets seem to be symmetrical but not public. Chip hijacking should be possible but no public known successes. |
| Xbox One/Series | ⭕ | ❓ | ❌ | Secrets not public. Plus no one cares about it enough except bootleggers. |

Basically the best bet is to implement a passthrough support. On RP2040 there's a PIO-based USB PHY that's work in progress. MAX3421E is always available as a second option too.

**Comment by dogtopus** [comment]
Also for PS4 native auth, it's preferred to use a dedicated core to handle RSA if you want to stay bare metal. Not sure why currently the secondary core is used for LEDs and display while they can be driven with little blocking if the IO to them is async and the display isn't doing full screen animations.

I'm currently adding support for RP2040 to my own DS4 controller library and will be able to post here for more info and auth algorithm example once I got the details ironed out. Meanwhile I do have a DS4 emulator in Python that has native auth capability so feel free to check the code [here](https://github.com/dogtopus/ffsds4/blob/master/ffsds4lib/ds4.py#L864).

**Comment by jbarket** [comment]
This is absolute gold. Thanks for sharing.

**Comment by TheTrainGoes** [comment]
Hey @dogtopus ,

Wanted to check in and see how progress on this went.

As a note - This project is no longer maintained. Please check out the Community Edition project. The Community Edition issues section can be found here: https://github.com/OpenStickFoundation/GP2040-CE/
